### PR TITLE
Implement rcutils_mkdir.

### PR DESCRIPTION
**Pull request by clalancette**
Needed for the spdlog logging backend.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

Connected to ros2/rcl_logging#17

### CODE REVIEW

**Comment by clairewangosrf** [comment]
(going through ci tutorial, test launch ci)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7764/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3860/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6347/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7611/)

**Comment by clalancette** [comment]
> Can you add some tests for this?

Done in 8075e9d

**Comment by clalancette** [comment]
New CI is in the meta-issue: https://github.com/ros2/ros2/issues/789#issuecomment-536019068

File: include/rcutils/filesystem.h [hunk 1]
```diff
@@ -144,6 +144,25 @@ rcutils_to_native_path(
...
+ *
+ * \param[in] abs_path
+ * \param[in] allocator
+ * \return bool True if making the directory was successful, False otherwise
+ *              False if path is NULL
+ *              False if path is empty
+ *              False if path is not absolute
+ *              False if any intermediate directories don't exist
+ */
+RCUTILS_PUBLIC
```
Comment (mjcarroll) [inline include/rcutils/filesystem.h]: Is is worth noting that this function is likely not thread safe?  I think most folks would assume that is the case, given the underlying call, but could probably bear repeating.

Comment (clalancette) [reply include/rcutils/filesystem.h]: Hm, interesting question.  It sort of depends on our definition of thread-safe.  It only uses stack variables and the parameter passed in, so there is no risk of memory corruption/memory races.  On the other hand, there is the directory race that `mkdir` is subject to (described in the openat documentation): http://man7.org/linux/man-pages/man2/open.2.html .  So from that perspective, it isn't thread-safe, but I'm not sure how we want to describe it.

File: include/rcutils/filesystem.h [hunk 2]
```diff
@@ -144,6 +144,25 @@ rcutils_to_native_path(
   const char * path,
   rcutils_allocator_t allocator);
 
+/// Create the specified directory.
+/**
+ * This function creates an absolutely-specified directory.  If any of the
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcutils/filesystem.h]: Thanks, fixed in 412dafb

File: include/rcutils/filesystem.h (hunk 1 above)
Comment (mjcarroll) [reply include/rcutils/filesystem.h]: I think leaving it at "This function is not thread safe" is probably suitable.  If someone is deeply concerned about it, then can consult the underlying code and documentation?

Comment (clalancette) [reply include/rcutils/filesystem.h]: I ended up adding that with a minor note to go look at the openat documentation for more information.

**Comment by clalancette** [comment]
All right, merging this one in as a pre-requisite to the spdlog logging framework.  Thanks for the reviews.

**Pull request merged**